sdk/src/main/scala/com/horizen/SidechainHistory.scala

```diff
@@ -41,7 +44,17 @@ class SidechainHistory private (val storage: SidechainHistoryStorage, params: Ne
 
   // Note: if block already exists in History it will be declined inside NodeViewHolder before appending.
   override def append(block: SidechainBlock): Try[(SidechainHistory, ProgressInfo[SidechainBlock])] = Try {
-    validators.map(_.validate(block, this)).foreach {
+    semanticBlockValidators.map(_.validate(block)).foreach {
```

Comment:
semanticBlockValidators -- make it as stream collection. We no longer will be forced to verify all validators if the first validator will be failed